LSPS2/README.md

```diff
@@ -0,0 +1,864 @@
+called `lsps3.getpromise` and the LSP responded to that call with a
+success (and the client has persisted the response), the flow
+continues:
+the client MUST NOT wait for the funding transaction to appear
+in its mempool, and MUST immediately send `channel_ready` at
+this point.
+
+As soon as both client and LSP have sent `channel_ready` for the
+new channel, the LSP then forwards payment parts to the client.
+In "no-MPP+var-invoice" mode there is only one payment part.
+
+The LSP generates non-standard forwards, where the amount received
+by the client is smaller than specified in the onion;
+the client MUST accept the non-standard forward(s), provided they
+sum to at least `payment_size_msat - opening_fee`.
+
+The LSP MUST ensure that each forwarded part forwards at least
+`htlc_minimum_msat` millisatoshis.
+The LSP MUST ensure that all parts forwarded sum up to at least
+`payment_size_msat - opening_fee`.
+
+For example, the LSP can use the algorithm below to determine
+how much to forward for each incoming payment part, while ensuring
+that both the above properties are preserved.
+
+* Put the payment parts in an array in arbitrary order.
+* For each part:
+  * Get the `part_amount`, the incoming amount for the part.
+  * If `part_amount - htlc_minimum_msat >= opening_fee`, then forward
+    `part_amount - opening_fee` for this part, then forward
+    the rest of the parts verbatim, if any.
+  * Else subtract `opening_fee = opening_fee - (part_amount - htlc_minimum_msat)`,
+    then forward `htlc_minimum_msat` for this part.
+* The above loop will not reach past the last part unless
+  the number of parts times `htlc_minimum_msat` exceeds the
+  `payment_size_msat`.
+  This case should be checked beforehand and cause the payment
+  to be failed due to too many payment parts.
+
+Once the client has received payment parts that sum up to at
+least `payment_size_msat`, it MUST claim at least one payment part.
+Once that is done, the protocol flow is considered complete.
+
+The parts that the client receives are non-standard, as the amount
+in the onion would not be the same as the amount sent by the LSP.
+Clients MUST suppress such checks for parts it recognizes as having
+been received to pay for a JIT Channel open.
+Clients would need some mechanism to recognize such payments, such as
+by flagging such payments in its invoice database, putting such
+payments in a separate invoice database, or adding special flags inside
+the `metadata` or `payment_secret` of the invoice, or by changing how
+`payment_secret` is computed based on whether the invoice is to pay for
+a JIT Channel or not and trying both ways.
+
+### 6.  Post-opening Normal Operation
+
+After the channel has been opened and an `alias` has been specified,
+the client MUST use the `alias` for future invoices on the
+channel, as normal for users of zero-confirmation channels.
+Note that use of `alias` forces the `alias` to always be used, even
+after the channel is confirmed.
```

Comment:
Can we set the `alias` to the same value as `jit_channel_scid` to avoid the mentioned multiple invoice issue?

Reply:
Not all implementations allow setting the `alias` of a new channel, sigh.  This is why there is text that the new channel may or may not have an `alias` that matches the `jit_channel_scid`.